### PR TITLE
Add ISP to structured metadata

### PR DESCRIPTION
**Pull request by relud**
This is needed in addition to #1117 for ISP to be preserved anywhere but `payload_bytes_raw`

### CODE REVIEW

File: ingestion-beam/pom.xml [hunk 1]
```diff
@@ -17,6 +17,21 @@
...
+            <dependencies>
+                <dependency>
+                    <groupId>${project.groupId}</groupId>
+                    <artifactId>ingestion-core</artifactId>
+                    <version>${project.version}</version>
+                </dependency>
+            </dependencies>
+        </profile>
+    </profiles>
+
```
Comment (jklukas) [inline ingestion-beam/pom.xml]: Is this block intended as part of this PR?

File: ingestion-beam/src/main/java/com/mozilla/telemetry/decoder/GeoIspLookup.java [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+  public class Fn extends SimpleFunction<PubsubMessage, PubsubMessage> {
+
+    private transient DatabaseReader ispReader;
+
+    private final Counter foundIp = Metrics.counter(GeoCityLookup.Fn.class, "found_ip");
+    private final Counter countIspAlreadyApplied = Metrics.counter(Fn.class, "isp_already_applied");
+    private final Counter foundIsp = Metrics.counter(Fn.class, "found_isp");
+    private final Counter countIpForwarded = Metrics.counter(GeoCityLookup.Fn.class,
+        "ip_from_x_forwarded_for");
+    private final Counter countIpRemoteAddr = Metrics.counter(GeoCityLookup.Fn.class,
```
Comment (jklukas) [inline ingestion-beam/src/main/java/com/mozilla/telemetry/decoder/GeoIspLookup.java:85]:
```suggestion
    private final Counter countIpRemoteAddr = Metrics.counter(GeoIspLookup.Fn.class,
```

File: ingestion-beam/src/main/java/com/mozilla/telemetry/decoder/GeoIspLookup.java [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+
+  @VisibleForTesting
+  public class Fn extends SimpleFunction<PubsubMessage, PubsubMessage> {
+
+    private transient DatabaseReader ispReader;
+
+    private final Counter foundIp = Metrics.counter(GeoCityLookup.Fn.class, "found_ip");
+    private final Counter countIspAlreadyApplied = Metrics.counter(Fn.class, "isp_already_applied");
+    private final Counter foundIsp = Metrics.counter(Fn.class, "found_isp");
+    private final Counter countIpForwarded = Metrics.counter(GeoCityLookup.Fn.class,
```
Comment (jklukas) [inline ingestion-beam/src/main/java/com/mozilla/telemetry/decoder/GeoIspLookup.java:83]:
```suggestion
    private final Counter countIpForwarded = Metrics.counter(GeoIspLookup.Fn.class,
```

File: ingestion-beam/src/main/java/com/mozilla/telemetry/decoder/GeoIspLookup.java [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+          //
+          // In practice, many of the "first" addresses are bogus or internal,
+          // so we target the immediate sending client IP.
+          String[] ips = xff.split("\\s*,\\s*");
+          ip = ips[Math.max(ips.length - 2, 0)];
+          countIpForwarded.inc();
+        } else {
+          ip = attributes.getOrDefault(Attribute.REMOTE_ADDR, "");
+          countIpRemoteAddr.inc();
+        }
```
Comment (jklukas) [inline ingestion-beam/src/main/java/com/mozilla/telemetry/decoder/GeoIspLookup.java:123]: I'd really like to avoid repeating this code for determining IP. We should break this up into:

```
            .apply(options.getInputType().read(options)) //
            .apply(ParseProxy.of()) // This would also now have responsibility for outputting an "ip" attribute
            .apply(GeoIspLookup.of(options.getGeoIspDatabase())) //
            .apply(GeoCityLookup.of(options.getGeoCityDatabase(), options.getGeoCityFilter())) // This would no longer strip IP address attributes
            .apply(DropIP.of()) // new transform that drops all IP address attributes
```

That probably should be a separate PR, though. This is fine for now.

File: ingestion-beam/pom.xml (hunk 1 above)
Comment (relud) [reply ingestion-beam/pom.xml]: nope, it got rebased out, i'm fixing tests and found that i was going to create hard-to-decipher conflicts with #1117

**Comment by relud** [comment]
copied @jklukas comments over to #1117, from which the relevant code originates.

closing in favor of https://github.com/scholtzan/gcp-ingestion/pull/1 which will merge this into #1117

**Comment by relud** [comment]
This change got lost from #1117, probably during a rebase somewhere, so I'm merging it based on existing r+'s
